python/ray/data/expressions.py

```diff
@@ -67,6 +68,8 @@ class Expr(ABC):
         subclasses like ColumnExpr, LiteralExpr, etc.
     """
 
+    return_dtype: DataType
```

Comment:
Generic expr doesn't have a return

```suggestion
    data_type: DataType
```